datafusion/physical-expr/src/window/aggregate.rs

```diff
@@ -85,6 +88,18 @@ impl PlainAggregateWindowExpr {
             );
         }
     }
+
+    fn is_window_constant(order_by: &LexOrdering, window_frame: &WindowFrame) -> bool {
```

Comment:
This is nice, I think we should have a comment to explain why
```suggestion
    // Returns true if every row in the partition has the same window frame. This allows 
    // for preventing bound + function calculation for every row due to the values being the 
    // same.
    //
    // This occurs when both bounds fall under either condition below:
    //  1. Bound is unbounded (`Preceding` or `Following`)
    //  2. Bound is `CurrentRow` while using `Range`units with no order by clause
    fn is_window_constant(order_by: &LexOrdering, window_frame: &WindowFrame) -> bool {
```

Reply:
TIL this convention in point two, I used to think this expr is invalid
```
2. Bound is `CurrentRow` while using `Range` units with no order by clause
```
I suggest to find some SQL specification link for this convention, and add to the comment. Maybe we should also double check postgres follows this behavior.